src/Community.VisualStudio.Toolkit.16.0/Community.VisualStudio.Toolkit.16.0.csproj

```diff
@@ -14,4 +17,13 @@
     <PackageReference Include="Microsoft.VisualStudio.Shell.Interop.12.1.DesignTime" Version="12.1.30329" />
   </ItemGroup>
 
+  <ItemGroup>
+    <!-- 
+    Prevent version conflicts with the `Microsoft.CodeAnalysis.*` libraries (caused by our reference to 
+    the analyzers project) by referencing a specific version here. We don't actually need this packages
+    to operate (and nor do any consumers of the toolkit), so we can safely exclude it from the output.
+    -->
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.11.0" ReferenceOutputAssembly="false" PrivateAssets="All" />
```

Comment:
ℹ️ I'd love to know if there's a better way to do this. Without adding this (and the equivalent in the project for `C.VS.T.17.0`), the compiler produces a warning because of the project reference to the analyzers project:

```
Found conflicts between different versions of "Microsoft.CodeAnalysis" that could not be resolved. [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
There was a conflict between "Microsoft.CodeAnalysis, Version=2.10.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" and "Microsoft.CodeAnalysis, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
    "Microsoft.CodeAnalysis, Version=2.10.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" was chosen because it was primary and "Microsoft.CodeAnalysis, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" was not. [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
    References which depend on "Microsoft.CodeAnalysis, Version=2.10.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" [D:\Cache\nuget\packages\microsoft.codeanalysis.common\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.dll]. [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Cache\nuget\packages\microsoft.codeanalysis.common\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Cache\nuget\packages\microsoft.codeanalysis.common\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.codeanalysis.common\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
    References which depend on "Microsoft.CodeAnalysis, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" []. [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.Analyzers\bin\Debug\netstandard1.3\Community.VisualStudio.Toolkit.Analyzers.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.Analyzers\bin\Debug\netstandard1.3\Community.VisualStudio.Toolkit.Analyzers.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.Analyzers\bin\Debug\netstandard1.3\Community.VisualStudio.Toolkit.Analyzers.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Cache\nuget\packages\microsoft.codeanalysis.csharp\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Cache\nuget\packages\microsoft.codeanalysis.csharp\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.codeanalysis.csharp\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Cache\nuget\packages\microsoft.codeanalysis.csharp.workspaces\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.Workspaces.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Cache\nuget\packages\microsoft.codeanalysis.csharp.workspaces\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.Workspaces.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.codeanalysis.csharp.workspaces\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.Workspaces.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Cache\nuget\packages\microsoft.codeanalysis.workspaces.common\2.10.0\lib\net46\Microsoft.CodeAnalysis.Workspaces.Desktop.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Cache\nuget\packages\microsoft.codeanalysis.workspaces.common\2.10.0\lib\net46\Microsoft.CodeAnalysis.Workspaces.Desktop.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.codeanalysis.workspaces.common\2.10.0\lib\net46\Microsoft.CodeAnalysis.Workspaces.Desktop.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Cache\nuget\packages\microsoft.codeanalysis.workspaces.common\2.10.0\lib\net46\Microsoft.CodeAnalysis.Workspaces.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Cache\nuget\packages\microsoft.codeanalysis.workspaces.common\2.10.0\lib\net46\Microsoft.CodeAnalysis.Workspaces.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.codeanalysis.workspaces.common\2.10.0\lib\net46\Microsoft.CodeAnalysis.Workspaces.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.Analyzers\bin\Debug\netstandard1.3\Community.VisualStudio.Toolkit.Analyzers.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.codeanalysis.csharp.workspaces\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.Workspaces.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.codeanalysis.workspaces.common\2.10.0\lib\net46\Microsoft.CodeAnalysis.Workspaces.Desktop.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Cache\nuget\packages\microsoft.visualstudio.texttemplating.15.0\16.0.28727\lib\net45\Microsoft.VisualStudio.TextTemplating.15.0.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Cache\nuget\packages\microsoft.visualstudio.texttemplating.15.0\16.0.28727\lib\net45\Microsoft.VisualStudio.TextTemplating.15.0.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.visualstudio.texttemplating.15.0\16.0.28727\lib\net45\Microsoft.VisualStudio.TextTemplating.15.0.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.visualstudio.texttemplating.vshost.15.0\16.0.28729\lib\net45\Microsoft.VisualStudio.TextTemplating.VSHost.15.0.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
Found conflicts between different versions of "Microsoft.CodeAnalysis.CSharp" that could not be resolved. [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
There was a conflict between "Microsoft.CodeAnalysis.CSharp, Version=2.10.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" and "Microsoft.CodeAnalysis.CSharp, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
    "Microsoft.CodeAnalysis.CSharp, Version=2.10.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" was chosen because it was primary and "Microsoft.CodeAnalysis.CSharp, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" was not. [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
    References which depend on "Microsoft.CodeAnalysis.CSharp, Version=2.10.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" [D:\Cache\nuget\packages\microsoft.codeanalysis.csharp\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.dll]. [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Cache\nuget\packages\microsoft.codeanalysis.csharp\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Cache\nuget\packages\microsoft.codeanalysis.csharp\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.codeanalysis.csharp\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
    References which depend on "Microsoft.CodeAnalysis.CSharp, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" []. [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.Analyzers\bin\Debug\netstandard1.3\Community.VisualStudio.Toolkit.Analyzers.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.Analyzers\bin\Debug\netstandard1.3\Community.VisualStudio.Toolkit.Analyzers.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.Analyzers\bin\Debug\netstandard1.3\Community.VisualStudio.Toolkit.Analyzers.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Cache\nuget\packages\microsoft.codeanalysis.csharp.workspaces\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.Workspaces.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Cache\nuget\packages\microsoft.codeanalysis.csharp.workspaces\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.Workspaces.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.codeanalysis.csharp.workspaces\2.10.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.Workspaces.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
        D:\Cache\nuget\packages\microsoft.visualstudio.texttemplating.15.0\16.0.28727\lib\net45\Microsoft.VisualStudio.TextTemplating.15.0.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
          Project file item includes which caused reference "D:\Cache\nuget\packages\microsoft.visualstudio.texttemplating.15.0\16.0.28727\lib\net45\Microsoft.VisualStudio.TextTemplating.15.0.dll". [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.visualstudio.texttemplating.15.0\16.0.28727\lib\net45\Microsoft.VisualStudio.TextTemplating.15.0.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
            D:\Cache\nuget\packages\microsoft.visualstudio.texttemplating.vshost.15.0\16.0.28729\lib\net45\Microsoft.VisualStudio.TextTemplating.VSHost.15.0.dll [D:\Code\GitHub\Community.VisualStudio.Toolkit\src\Community.VisualStudio.Toolkit.16.0\Community.VisualStudio.Toolkit.16.0.csproj]
```

The `14.0` and `15.0` projects don't have this warning, but the `16.0` and `17.0` projects do. I tried a bunch of different options for generating binding redirects, but nothing seemed to make a difference. I've set `ReferenceOutputAssembly` to false and `PrivateAssets` to "all" to prevent this package from ending up as a dependency in the NuGet package, but it still seems like you can use the types from `Microsoft.CodeAnalysis.*` in the toolkit project, which isn't ideal.